### PR TITLE
Feature/backend resilience

### PR DESCRIPTION
**Pull request by marcel-vesely-kinit**
- Backend resilience (checking REANA availability, more robust image building, checking for existence of docker image when running an experiment run)
- Frontend changes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -162,8 +162,11 @@ cython_debug/
 
 # Visual Studio Code
 .vscode/
-backend/.vscode/
-.vs/
+# !.vscode/settings.json
```
Comment (andrejridzik) [inline .gitignore:165]: No need to update this I guess

File: backend/app/config.py [hunk 2]
```diff
@@ -13,7 +13,8 @@
 CHECK_REANA_CONNECTION_INTERVAL = 60
 RUN_TEMP_OUTPUT_FOLDER = "output-temp"
 RUN_OUTPUT_FOLDER = "output"
-REPOSITORY_NAME = "rail-exp-templates"
+# REPOSITORY_NAME = "rail-exp-templates" #TODO
```
Comment (andrejridzik) [inline backend/app/config.py:16]: Did you forget to remove this?

File: backend/app/services/experiment.py [hunk 3]
```diff
@@ -164,36 +138,115 @@ async def schedule_image_building(self) -> None:
...
-
-                await new_exp_run.create()
-                await experiment_run.replace()
-                await self.add_run_to_execute(new_exp_run.id)
-            else:
-                experiment_run.update_state(RunState.CRASHED)
-                await experiment_run.replace()
+            try:
+                experiment_run, experiment = await self.init_run(exp_run_id)
+                if experiment_run is None:
```
Comment (andrejridzik) [inline backend/app/services/experiment.py:143]: What does this case mean? Don't we need to do some action here?

Comment (marcel-vesely-kinit) [reply backend/app/services/experiment.py:143]: necessary actions are performed within the `init_run` function. And this use case describes a situation in which we firstly couldn't find a docker image for corresponding template and we further didnt succeed in rebuilding it either

Comment (andrejridzik) [reply backend/app/services/experiment.py:143]: Should we at least log something?

Comment (marcel-vesely-kinit) [reply backend/app/services/experiment.py:143]: sure, it wont hurt

**Pull request merged**